src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -294,6 +294,71 @@ public Mnemonic CreateWallet(string password, string name, string passphrase, Mn
             return mnemonic;
         }
 
+        /// <inheritdoc />
+        public string SignMessage(string password, string walletName, string externalAddress, string message)
+        {
+            Guard.NotEmpty(password, nameof(password));
+            Guard.NotEmpty(walletName, nameof(walletName));
+            Guard.NotEmpty(message, nameof(message));
+            Guard.NotEmpty(externalAddress, nameof(externalAddress));
+
+            string signature = string.Empty;
+
+            // Get wallet
+            Wallet wallet = this.GetWalletByName(walletName);
+
+            // Check the password.
+            try
+            {
+                if (!wallet.IsExtPubKeyWallet)
+                    Key.Parse(wallet.EncryptedSeed, password, wallet.Network);
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogTrace("Exception occurred: {0}", ex.ToString());
+                this.logger.LogTrace("(-)[EXCEPTION]");
+                throw new SecurityException(ex.Message);
+            }
+
+            // Sign the message.
+            try
+            {
+                HdAddress hdAddress = wallet.FindHDAddressByExternalAddress(externalAddress);
+                Key privateKey = wallet.GetExtendedPrivateKeyForAddress(password, hdAddress).PrivateKey;
+                BitcoinSecret secret = this.network.CreateBitcoinSecret(privateKey);
+                signature = secret.PrivateKey.SignMessage(message);
+            }
+            catch (Exception ex)
```

Comment:
I don't think returning an empty string is a good idea here.
If the method fails, a log is gonna be printed but then an empty string will be returned to the controller, and the user will not know what happened.
Probably better to remove the try catch here and let the error be thrown out.